### ISSUE TITLE
[Bug]: Git-Sync shows uncommited changes (generated by system) frequently 

### ISSUE THREAD

**Issue by pranavkanade** [appsmithorg/appsmith#19877]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

A user complained that even though they always commit all the changes in the app. They frequently see 1/2 uncommitted changes pending. 

### Steps To Reproduce

user's repro - 
```
We're on Appsmith Business v1.9.3.1, this morning when I can back to the master branch I had 2 changes apparently, it just said 2 pages modified which is fine when you know you made changes but yes your point on showing exactly what they are would be ideal as we're not expecting to see any changes. I discarded and pulled and was fine again.
```

### Version

v1.9.3.1

**Comment by ame-appsmith** [comment]
[Another user](https://app.frontapp.com/open/cnv_13wqf1l8?key=XbQRaPeTW-qjoFi0cswL3c1EHf2NgX1s) complains about this issue.

**Comment by MAL0780** [comment]
Another user complains about this issue. It is making me crazy!
Please, prioritize.
Tnx

**Comment by infinitetrooper** [comment]
Being solved in #27055

**Comment by infinitetrooper** [comment]
Being closed in epic #24793
